pyk/src/pyk/ktool/kprove.py

```diff
@@ -3,41 +3,82 @@
 import subprocess
 import sys
 from pathlib import Path
-from typing import List
-
-from .cli_utils import fatal, gen_file_timestamp, notif
-from .kast import (
+from subprocess import CompletedProcess, run
+from typing import Iterable, List, Optional
+
+from ..cli_utils import (
+    check_dir_path,
+    check_file_path,
+    fatal,
+    gen_file_timestamp,
+    notif,
+)
+from ..kast import (
     KAst,
     KDefinition,
     KFlatModule,
     KImport,
     KRequire,
-    buildSymbolTable,
     flattenLabel,
-    prettyPrintKast,
-    readKastTerm,
 )
-from .prelude import mlTop
-from .utils import hash_str
+from ..prelude import mlTop
+from .kprint import KPrint
 
 
-class KPrint:
-    """Given a kompiled directory, build an unparser for it.
-    """
+def kprovex(
+    spec_file: Path,
+    *,
+    directory: Optional[Path] = None,
+    include_dirs: Iterable[Path] = (),
+    emit_json_spec: Optional[Path] = None,
+    dry_run=False,
+) -> None:
+    check_file_path(spec_file)
 
-    def __init__(self, kompiledDirectory):
-        self.kompiledDirectory = Path(kompiledDirectory)
-        self.definition = readKastTerm(self.kompiledDirectory / 'compiled.json')
-        self.symbolTable = buildSymbolTable(self.definition, opinionated=True)
-        self.definitionHash = hash_str(self.definition)
+    for include_dir in include_dirs:
+        check_dir_path(include_dir)
 
-    def prettyPrint(self, kast, debug=False):
-        """Given a KAST term, pretty-print it using the current definition.
+    args = _build_arg_list(
+        directory=directory,
+        include_dirs=include_dirs,
+        dry_run=dry_run,
+        emit_json_spec=emit_json_spec,
+    )
 
-        -   Input: KAST term in JSON.
-        -   Output: Best-effort pretty-printed representation of the KAST term.
-        """
-        return prettyPrintKast(kast, self.symbolTable, debug=debug)
+    proc_res = _kprovex(str(spec_file), *args)
+
+    if proc_res.returncode:
+        raise RuntimeError(f'Command kprovex failed for: {spec_file}')
+
+
+def _build_arg_list(
+    *,
+    directory: Optional[Path],
+    include_dirs: Iterable[Path],
+    emit_json_spec: Optional[Path],
+    dry_run: bool,
+) -> List[str]:
+    args = []
+
+    if directory:
+        args += ['--directory', str(directory)]
+
+    for include_dir in include_dirs:
+        args += ['-I', str(include_dir)]
+
+    if emit_json_spec:
+        args += ['--emit-json-spec', str(emit_json_spec)]
+
+    if dry_run:
+        args.append('--dry-run')
+
+    return args
+
+
+def _kprovex(spec_file: str, *args: str) -> CompletedProcess:
+    run_args = ['kprovex', spec_file] + list(args)
+    notif(' '.join(run_args))
+    return run(run_args, capture_output=True)
 
 
 class KProve(KPrint):
```

Comment:
We should probably have class `KProve` make use of function `kprovex` I think?

Reply:
That's tricky, because some clients rely on updating `KProve.prover`, e.g.: https://github.com/runtimeverification/erc20-verification/blob/master/ksummarize/__main__.py#L197.